spec/classes/init_spec.rb

```diff
@@ -260,6 +260,48 @@
 
   end
 
+  context 'with selboolean specified as a hash' do
+    let(:facts) { { :osfamily => 'RedHat' } }
+    let :params do
+      {
+        :selboolean_hiera_merge => 'false',
+        :selboolean => {
+          'nfs_export_all_ro' => {
+            'value' => 'on',
+          },
+          'nfs_export_all_rw' => {
+            'persistent' => true,
+            'value'      => 'on',
+          },
+        }
+      }
+    end
+
+    it { should contain_class('types') }
+    it {
+      should contain_selboolean('nfs_export_all_ro').with({
+        'value' => 'on',
+      })
+    }
+    it {
+      should contain_selboolean('nfs_export_all_rw').with({
+        'persistent' => 'true',
+        'value'      => 'on',
+      })
+    }
+  end
+
+  context 'with selboolean specified as an invalid type' do
+    let(:facts) { { :osfamily => 'RedHat' } }
+    let(:params) { { :selboolean => ['not','a','hash'] } }
+
+    it 'should fail' do
+      expect {
+        should contain_class('types')
+      }.to raise_error(Puppet::Error,/\["not", "a", "hash"\] is not a Hash\./)
```

Comment:
`/is not a Hash/` would be ok here. We use the shorter ones because the values changed quite a bit between puppet v3 and v4